pkg/api/validation/validation.go

```diff
@@ -735,6 +743,32 @@ func validateConfigMapVolumeSource(configMapSource *api.ConfigMapVolumeSource, f
 	return allErrs
 }
 
+func validateHugePagesVolumeSource(hugePages *api.HugePagesVolumeSource, fldPath *field.Path) field.ErrorList {
+	allErrs := field.ErrorList{}
+
+	// TODO(pprokop): add support for other PageSizes
+	if hugePages.PageSize != "2M" {
+		allErrs = append(allErrs, field.Invalid(fldPath.Child("pageSize"), hugePages.PageSize, "Not supported page size"))
+	}
+	maxSize, err := resource.ParseQuantity(hugePages.MaxSize)
```

Comment:
It is an interesting case because doing mount with option size=5.3M will not fail but it will be transformed into 5 bytes because kernel will discard everything after dot. The code is [here](https://github.com/torvalds/linux/blob/master/fs/hugetlbfs/inode.c#L1083). So either K8s should not allowed using fractional sizes or convert every size option  to bytes before mounting.